### ISSUE TITLE
A way to report invalid NYC configuration.

### ISSUE THREAD

**Issue by coreyfarrell** [istanbuljs/nyc#966]
An automated method should be able to report errors in NYC configuration.  For example if someone has `{"extensions": [".mjs"]}` nyc is not going to act as they expect.  This ultimately causes support requests to be posted a bug reports.

I'm not sure where this check should happen.  We could have NYC check the configuration at startup and error out if unknown keys are specified.

**Comment by LEQADA** [comment]
Hi @coreyfarrell and @JaKXz, I could take it over. Yargs has a [`strict` mode](https://github.com/yargs/yargs/blob/8792d13445458c51bdff6612b1edda5aa344b6a0/docs/api.md#strictenabledtrue). I'd use it. Is it enough, wdyt?

**Comment by coreyfarrell** [comment]
I don't believe yargs strict mode can be used by nyc.  The problem is that sub-commands can have different options.  For example the main command supports `--hook-run-in-context`, but this is not supported by `nyc report`.  So if someone had `hookRunInContext: true` in their NYC configuration yargs strict mode would error out on `nyc report` because hookRunInContext is unknown to that sub-command.  I'd appreciate any help or ideas you have for this.  I'm leaning towards wanting to put any config/option verification logic into https://github.com/istanbuljs/load-nyc-config/ so it can be used by our babel plugin.  That module does not depend on yargs and I'd prefer to keep it that way.  I'm hoping to do more work on the load-nyc-config module soon, unfortunately I don't have any time currently.
